source/Octopus.Tentacle.Tests.Integration/TentacleCommandLineTests.cs

```diff
@@ -0,0 +1,659 @@
+  }}
+}}
+");
+            }
+            else
+            {
+                stdout.Should().Be($@"{{
+  ""Octopus"": {{
+    ""Home"": ""{JsonFormattedPath(clientAndTentacle.RunningTentacle.HomeDirectory)}"",
+    ""Watchdog"": {{
+      ""Enabled"": false,
+      ""Instances"": ""*"",
+      ""Interval"": 0
+    }}
+  }},
+  ""Tentacle"": {{
+    ""CertificateThumbprint"": ""{clientAndTentacle.RunningTentacle.Thumbprint}"",
+    ""Communication"": {{
+      ""TrustedOctopusServers"": [
+        {{
+          ""Thumbprint"": ""{clientAndTentacle.Server.Thumbprint}"",
+          ""CommunicationStyle"": 1,
+          ""KubernetesTentacleCommunicationMode"": {{
+            ""Value"": ""Polling""
+          }},
+          ""Address"": null,
+          ""Squid"": null,
+          ""SubscriptionId"": null
+        }}
+      ]
+    }},
+    ""Deployment"": {{
+      ""ApplicationDirectory"": ""{JsonFormattedPath(clientAndTentacle.RunningTentacle.ApplicationDirectory)}""
+    }},
+    ""Services"": {{
+      ""ListenIP"": null,
+      ""NoListen"": false,
+      ""PortNumber"": {clientAndTentacle.RunningTentacle.ServiceUri.Port}
+    }}
+  }}
+}}
+");
+            }
+
+            await Task.CompletedTask;
+        }
+
+        [Test]
+        [TentacleConfigurations(testCurrentVersionOnly: true)]
+        [WindowsTest]
+        [NonParallelizable]
+        public async Task WatchdogCreateAndDeleteCommand(TentacleConfigurationTestCase tc)
+        {
+            await using var clientAndTentacle = await tc.CreateBuilder().Build(CancellationToken);
+            var create = await RunCommandWithValidExitCodeAssert(tc, $"watchdog --create --instances={clientAndTentacle.RunningTentacle.InstanceName}");
+            create.StdError.Should().BeNullOrEmpty();
+            create.StdOut.Should().ContainEquivalentOf("Creating watchdog task");
+            var delete = await RunCommandWithValidExitCodeAssert(tc, $"watchdog --delete --instances={clientAndTentacle.RunningTentacle.InstanceName}");
+            delete.StdError.Should().BeNullOrEmpty();
+            delete.StdOut.Should().ContainEquivalentOf("Removing watchdog task");
+        }
+        
+        FileVersionInfo GetVersionInfo(TentacleConfigurationTestCase tentacleConfigurationTestCase)
+        {
+            var tentacleExe = TentacleExeFinder.FindTentacleExe(tentacleConfigurationTestCase.TentacleRuntime);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return FileVersionInfo.GetVersionInfo(tentacleExe);
+            }
+
+            //todo: change this to trust the value set in the context.TentacleExePath (will need a renovation of ExePathResolver to be non windows specific)
+            return FileVersionInfo.GetVersionInfo($"{tentacleExe}.dll");
+        }
+
+        async Task<(int ExitCode, string StdOut, string StdError)> RunCommandWithValidExitCodeAssert(TentacleConfigurationTestCase tentacleConfigurationTestCase, string input)
```

Comment:
nit: Maybe rename this to `RunCommandAndAssertExitsWithSuccessExitCode`  successful.

Reply:
Done